hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -955,19 +964,24 @@ private void bulkLoadHFile(HStoreFile sf) throws IOException {
       if (!result.isEmpty()) {
         // initialize the thread pool for closing store files in parallel.
         ThreadPoolExecutor storeFileCloserThreadPool = this.region
-            .getStoreFileOpenAndCloseThreadPool("StoreFileCloser-"
-              + this.region.getRegionInfo().getEncodedName() + "-" + this.getColumnFamilyName());
-
+            .getStoreFileOpenAndCloseThreadPool();
+        String executingThreadNamePrefix = "StoreFileCloser-"
+          + this.region.getRegionInfo().getEncodedName() + "-" + this.getColumnFamilyName();
+        AtomicInteger threadId = new AtomicInteger(0);
         // close each store file in parallel
         CompletionService<Void> completionService =
           new ExecutorCompletionService<>(storeFileCloserThreadPool);
         for (HStoreFile f : result) {
           completionService.submit(new Callable<Void>() {
             @Override
             public Void call() throws IOException {
+              Thread t = Thread.currentThread();
```

Comment:
All these looks bit hacky

Reply:
Changing the thread name is not necessary, but it will be useful when jstack  the process, we will know something in detail.Yeah, it's bit hacky

> All these looks bit hacky

Changing the thread name is not necessary, but it will be useful when jstack  the process, we will know something in detail.Yeah, it's bit hacky

Reply:
> All these looks bit hacky

What do you think about here, Anoop? Just remove the name related logic? Or just set and not reset ?

Reply:
I dont think we should do these hacks.  The threads can be named with regionName and some seqNo.  Anyway even if we do these hacks and add the cf name, it says till that level only. Still from jstack we can not know which file open it is stuck with (if assume we took it when there was a stuck case).  We can make sure we have proper log which says opening store file xxx for cf x and that log will have to say the thread name also.  So later with jstack and this log we can track down issues.  Sounds ok?

Reply:
> I dont think we should do these hacks. The threads can be named with regionName and some seqNo. Anyway even if we do these hacks and add the cf name, it says till that level only. Still from jstack we can not know which file open it is stuck with (if assume we took it when there was a stuck case). We can make sure we have proper log which says opening store file xxx for cf x and that log will have to say the thread name also. So later with jstack and this log we can track down issues. Sounds ok?

Yeah, definitely, let me update the patch soon.